### PR TITLE
Improve payne hanek

### PR DESCRIPTION
**Pull request by shibatch**
This patch improves the Payne-Hanek reduction algorithm.
The restriction on the input domain of trig functions is all removed.

### CODE REVIEW

File: src/libm-tester/tester2simddp.c [hunk 1]
```diff
@@ -88,16 +88,12 @@ typedef Sleef___m512d_2 vdouble2;
...
-#define CONFIG 1
-#include "helpervecext.h"
-#include "norename.h"
-#endif
-
-#ifdef ENABLE_PUREC
-#define CONFIG 1
-#include "helperpurec.h"
-#include "norename.h"
+#ifdef ENABLE_AVX512FNOFMA
```
Comment (fpetrogalli-arm) [inline src/libm-tester/tester2simddp.c]: The changes in this file seem to be unnecessary. If I understand correctly, these are the "nofma" and "pure_c*_scalar" target that are needed for the "determinism" patch, and are unrelated to the improvements of the payne hanek algorithm.

In fact, "rename*nofma.h"  is not included in this patch.

May I ask you to revert the changes in this file?

File: src/libm-tester/tester2simddp.c [hunk 2]
```diff
@@ -126,6 +112,18 @@ typedef Sleef_vector_double_2 vdouble2;
 typedef Sleef_vector_float_2 vfloat2;
 #endif
 
+#ifdef ENABLE_VECEXT
```
Comment (fpetrogalli-arm) [inline src/libm-tester/tester2simddp.c]: Please revert the change in line 115-126.

File: src/libm-tester/tester2simddp.c [hunk 3]
```diff
@@ -111,11 +99,9 @@ typedef Sleef_float32x4_t_2 vfloat2;
 #ifdef ENABLE_SVE
 #define CONFIG 1
 #include "helpersve.h"
-#ifdef DORENAME
```
Comment (fpetrogalli-arm) [inline src/libm-tester/tester2simddp.c]: Why remove this?

File: src/libm-tester/tester2simdsp.c [hunk 4]
```diff
@@ -127,6 +121,12 @@ typedef Sleef_vector_double_2 vdouble2;
 typedef Sleef_vector_float_2 vfloat2;
 #endif
 
+#ifdef ENABLE_PUREC
```
Comment (fpetrogalli-arm) [inline src/libm-tester/tester2simdsp.c]: Please restore line 124-129 in the original position.

File: src/libm/sleefsimdsp.c [hunk 5]
```diff
@@ -2473,6 +2463,7 @@ EXPORT CONST vfloat __tgammaf_u1_finite(vfloat)         __attribute__((weak, ali
 // gcc -DENABLE_MAIN -Wno-attributes -I../common -I../arch -DENABLE_AVX2 -mavx2 -mfma sleefsimdsp.c ../common/common.c -lm
 #include <stdio.h>
 #include <stdlib.h>
+#include <math.h>
```
Comment (fpetrogalli-arm) [inline src/libm/sleefsimdsp.c]: Spurious include?

File: src/libm-tester/tester2simddp.c (hunk 3 above)
Comment (shibatch) [reply src/libm-tester/tester2simddp.c]: Because DORENAME is always defined.
